spyder/plugins/pylint/tests/test_pylint.py

```diff
@@ -335,5 +336,25 @@ def test_custom_interpreter(pylint_plugin, tmp_path, qtbot,
         assert errors
 
 
+def test_get_environment(mocker):
+    """Test that the environment variables depend on the OS."""
+    if os.name == 'nt':
+        mocker.patch("spyder.plugins.pylint.main_widget.is_conda_based_app",
+                     return_value=False)
+        mocker.patch("spyder.plugins.pylint.main_widget.is_anaconda",
+                     return_value=False)
+        mocker.patch("spyder.plugins.pylint.main_widget.get_home_dir",
+                     return_value='')
+
+    etalon = {
+        "nt": ["APPDATA", "PYTHONIOENCODING", "PYTHONPATH", "USERPROFILE"],
+        "posix": ["PYTHONIOENCODING", "PYTHONPATH"]}
```

Comment:
```suggestion
    expected_vars = {
        "nt": ["APPDATA", "PYTHONIOENCODING", "PYTHONPATH", "USERPROFILE"],
        "posix": ["PYTHONIOENCODING", "PYTHONPATH"]
    }
```

I don't understand what `etalon` means.